File: core/src/com/broll/gainea/server/core/cards/impl/play/C_Secrets.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.Card;
import com.broll.gainea.server.core.goals.AbstractGoal;
import com.broll.gainea.server.core.player.Player;
import com.broll.gainea.server.core.utils.MessageUtils;

import java.util.List;

public class C_Secrets extends Card {
    public C_Secrets() {
        super(40, "Bestechung", "Schaut euch die Ziele und Aktionskarten eines beliebigen Spielers an");
        setDrawChance(0.8f);
    }

    @Override
    public boolean isPlayable() {
        return true;
    }

    @Override
    protected void play() {
        Player player = selectHandler.selectOtherPlayer(owner, "Welchen Spieler bestechen?");
        List<Card> cards = player.getCardHandler().getCards();
        List<AbstractGoal> goals = player.getGoalHandler().getGoals();
        String name = player.getServerPlayer().getName();
        String text = name + "'s Ziele: \n";
        for (AbstractGoal goal : goals) {
            text += " - " + goal.getText() + " ( " +goal.getDifficulty().getPoints()+"P "  + goal.getRestrictionInfo() + ")\n";
        }
        text += name + "'s Aktionskarten: \n";
        for (Card card : cards) {
            text += " - " + card.getTitle() + "\n";
        }
        MessageUtils.displayMessage(game, owner, text);
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_Secrets.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.utils.displayConfirmMessage

class C_Secrets : Card(40, "Bestechung", "Schaut euch die Ziele und Aktionskarten eines beliebigen Spielers an") {
    init {
        drawChance = 0.8f
    }

    override val isPlayable: Boolean
        get() = true

    override fun play() {
        val player = selectHandler.selectOtherPlayer(owner, "Welchen Spieler bestechen?")
        val cards = player.cardHandler.cards
        val goals = player.goalHandler.goals
        val name = player.serverPlayer.name
        var text = "$name's Ziele: \n"
        for (goal in goals) {
            text += " - " + goal.text + " ( " + goal.difficulty.points + "P " + goal.restrictionInfo + ")\n"
        }
        text += "$name's Aktionskarten: \n"
        for (card in cards) {
            text += " - " + card.title + "\n"
        }
        owner.displayConfirmMessage(text)
    }
}
